File: src/JavaScript/Pyramid-Array.js
/**
 * @module Pyramid-Array
 * @file JS 中的 repeat 可以考虑 Array(n).fill()
 */

const repeat = require("./toolkit/Vector").repeat;

function pyramid(n) {
  return [...Array(n)].map((_, i) => repeat([1], i + 1));
}

console.log(pyramid(3));

File: src/Julia/Pyramid-Array.jl

using Test

function pyramid(n::Int64)::Vector{Vector}
    # 等价的三种写法
    # [ones(Int, k) for k ∈ 1:n]
    # [fill(1, k) for k ∈ 1:n]
    [repeat([1], k) for k ∈ 1:n]
end

@test pyramid(3) == [[1], [1, 1], [1, 1, 1]]